packages/formatters/src/formatters.ts

```diff
@@ -336,11 +336,66 @@ export function formatEmoji<EmojiId extends Snowflake>(
 	animated?: boolean,
 ): `<:_:${EmojiId}>` | `<a:_:${EmojiId}>`;
 
-export function formatEmoji<EmojiId extends Snowflake>(
+/**
+ * Formats an emoji id and name into a fully qualified emoji identifier.
+ *
+ * @typeParam EmojiId - This is inferred by the supplied emoji id
+ * @param emojiId - The emoji id to format
+ * @param animated - Whether the emoji is animated
+ * @param name - The name of the emoji
+ */
+export function formatEmoji<EmojiId extends Snowflake, Name extends string>(
 	emojiId: EmojiId,
-	animated = false,
-): `<:_:${EmojiId}>` | `<a:_:${EmojiId}>` {
-	return `<${animated ? 'a' : ''}:_:${emojiId}>`;
+	animated: boolean,
+	name: Name,
+): `<:${Name}:${EmojiId}>` | `<a:${Name}:${EmojiId}>`;
+
+/**
+ * Formats an emoji id and name into a fully qualified emoji identifier.
+ *
+ * @typeParam FormatEmojiOptions - An options object to format emoji.
+ */
+export function formatEmoji<EmojiId extends Snowflake, Name extends string>({
+	emojiId,
+	animated,
+	name,
+}: Required<FormatEmojiOptions<EmojiId, Name>>): `<:${Name}:${EmojiId}>` | `<a:${Name}:${EmojiId}>`;
```

Comment:
You can just specify `options`:

```suggestion
export function formatEmoji<EmojiId extends Snowflake, Name extends string>(
	options: Required<FormatEmojiOptions<EmojiId, Name>>,
): `<:${Name}:${EmojiId}>` | `<a:${Name}:${EmojiId}>`;
```